### PR TITLE
tiltfile: add os.getenv, os.putenv, os.unsetenv. Fixes https://github.com/windmilleng/tilt/issues/2564

### PR DESCRIPTION
**Pull request by nicks**
Hello @jazzdan,

Please review the following commits I made in branch nicks/ch6958/putenv:

fcf4b19a168d47b8953667637b877ea1ac1e9fff (2020-05-12 17:01:51 -0400)
tiltfile: add os.getenv, os.putenv, os.unsetenv. Fixes https://github.com/windmilleng/tilt/issues/2564

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
actually, this isn't quite right, one sec

**Comment by nicks** [comment]
ok, fixed.

I am...not happy about how much code I needed to duplicate from starlark to get this to work right. The core problem is that we need to make sure updates to the dict are reflected in the environment. The code in environ_methods is basically just copied-and-pasted mechanically, where I just replaced *Dict with Environ (since they use the same API)

**Comment by nicks** [comment]
(There's obviously a simpler implementation possible where *Dict supports different backing stores, but I'm not sure starlark would accept that upstream)

File: internal/tiltfile/os/os.go [hunk 1]
```diff
@@ -77,18 +85,70 @@ func osName() string {
...
-		err := result.SetKey(starlark.String(pair[0]), starlark.String(pair[1]))
-		if err != nil {
-			return nil, err
-		}
+func getenv(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var varname starlark.Value
+	var defaultVal starlark.Value = starlark.None
+	err := starkit.UnpackArgs(t, fn.Name(), args, kwargs,
+		"varname", &varname,
+		"value?", &defaultVal,
```
Comment (jazzdan) [inline internal/tiltfile/os/os.go:93]:
```suggestion
		"default?", &defaultVal,
```

File: internal/tiltfile/os/os.go [hunk 2]
```diff
@@ -77,18 +85,70 @@ func osName() string {
...
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	varnameStr, ok := value.AsString(varname)
+	if !ok {
+		return nil, fmt.Errorf("varname must be a string, actual: %s", varname)
```
Comment (jazzdan) [inline internal/tiltfile/os/os.go:101]: Does defaultVal also need to be enforced to be a string?

File: internal/tiltfile/os/os.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/os/os.go:93]: ahhhh...good catch! this doc
https://docs.python.org/2/library/os.html
says it's value but this doc says it's default
https://docs.python.org/3/library/os.html
and when i tried it out in a real python interpreter, they all only accepted 'default' :rolledeyes:

File: internal/tiltfile/os/os.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/os/os.go:101]: surprisingly no!

```
python -c "import os; print(os.getenv('x', {}))"
{}
```

File: internal/tiltfile/os/os.go (hunk 1 above)
Comment (jazzdan) [reply internal/tiltfile/os/os.go]: Weird. Windows weird.

**Pull request merged**